tests/test_part_multiple_rotate.py

```diff
@@ -26,57 +26,54 @@ def test_rotate():
     assembly.add(mypart6)
     assembly.add(mypart7)
     assembly.add(mypart8)
-    assembly.add(mypart9) 
-
+    assembly.add(mypart9)
 
     # Test that a part is rotated multiple times.
     assembly.rotateFreezeLeft(mypart1)
     assembly.rotateFreezeFront(mypart1)
     assembly.rotateFreezeTop(mypart1)
     assembly.rotateFreezeLeft(mypart1)
     assembly.rotateFreezeLeft(mypart1)
-    
+
     assembly.rotateFreezeLeft(mypart2)
     assembly.rotateFreezeFront(mypart2)
     assembly.rotateFreezeTop(mypart2)
-    
+
     assembly.rotateFreezeLeft(mypart3)
     assembly.rotateFreezeFront(mypart3)
-    
+
     assembly.rotateFreezeFront(mypart4)
     assembly.rotateFreezeLeft(mypart4)
-    
+
     assembly.rotateFreezeTop(mypart5)
     assembly.rotateFreezeFront(mypart5)
-    
+
     assembly.rotateFreezeFront(mypart6)
     assembly.rotateFreezeTop(mypart6)
-    
+
     assembly.rotateFreezeLeft(mypart7)
     assembly.rotateFreezeTop(mypart7)
-    
+
     assembly.rotateFreezeTop(mypart8)
     assembly.rotateFreezeLeft(mypart8)
-    
+
     assembly.rotateFreezeLeft(mypart9)
     assembly.rotateFreezeFront(mypart9)
     assembly.rotateFreezeTop(mypart9)
     assembly.rotateFreezeLeft(mypart9)
     assembly.rotateFreezeFront(mypart9)
     assembly.rotateFreezeTop(mypart9)
-    
-    
-    
+
     assembly_def = assembly.export()
-    assert assembly_def[0]["rotate"] == [180, 0, 270]
-    assert assembly_def[1]["rotate"] == [180, 0, 90]
-    assert assembly_def[2]["rotate"] == [90, 0, 90]
-    assert assembly_def[3]["rotate"] == [0, 90, 90]
-    assert assembly_def[4]["rotate"] == [90, 0, 90]
-    assert assembly_def[5]["rotate"] == [90, 90, 0]
-    assert assembly_def[6]["rotate"] == [90, 90, 0]
-    assert assembly_def[7]["rotate"] == [0, 90, 90]
-    assert assembly_def[8]["rotate"] == [270, 0, 270]
+    assert assembly_def["parts"][0]["rotate"] == [0, 1, 2, 0, 0]
+    assert assembly_def["parts"][1]["rotate"] == [0, 1, 2]
+    assert assembly_def["parts"][2]["rotate"] == [0, 1]
+    assert assembly_def["parts"][3]["rotate"] == [1, 0]
+    assert assembly_def["parts"][4]["rotate"] == [2, 1]
+    assert assembly_def["parts"][5]["rotate"] == [1, 2]
+    assert assembly_def["parts"][6]["rotate"] == [0, 2]
+    assert assembly_def["parts"][7]["rotate"] == [2, 0]
+    assert assembly_def["parts"][8]["rotate"] == [0, 1, 2, 0, 1, 2]
```

Comment:
As mentioned in another PR. This should be a list of dictionaries describing how the part was rotated. A list of magic numbers are dangerous. 

e.g.
part["rotate"] = [{'side': 'TOP', 'angle': 90}, {'side': "LEFT", "angle": 90}]

Passing angle to engines future proof the assembly.

Reply:
SS162